mysql-test/main/rpl_mysqldump_slave.test

```diff
@@ -83,6 +83,52 @@ DROP TABLE t2;
 --replace_regex /MASTER_LOG_POS=[0-9]+/MASTER_LOG_POS=BINLOG_START/
 --exec $MYSQL_DUMP_SLAVE --compact --master-data --single-transaction --gtid test
 
+#
+# MDEV-32611 Added test for mysqldump --delete-master-logs option.
+#
+
+connection master;
 
+CREATE TABLE t (
+    id int
+);
+
+insert into t values (1);
+insert into t values (2);
+
+--let $predump_binlog_filename= query_get_value(SHOW MASTER STATUS, File, 1)
+--let $predump_binlog_pos= query_get_value(SHOW MASTER STATUS, Position, 1)
+
+# Execute mysqldump with delete-master-logs option
+--replace_regex /MASTER_LOG_POS=[0-9]+/MASTER_LOG_POS=BINLOG_START/
+--exec $MYSQL_DUMP --compact --no-create-info --no-data --delete-master-logs test
+
+--let $postdump_binlog_filename= query_get_value(SHOW MASTER STATUS, File, 1)
+--let $postdump_binlog_pos= query_get_value(SHOW MASTER STATUS, Position, 1)
+
+--let $postdump_first_binary_log_filename= query_get_value(SHOW BINARY LOGS, Log_name, 1)
+
+if ($predump_binlog_filename == $postdump_binlog_filename)
+{
+  --echo # predump_binlog_filename: $predump_binlog_filename
+  --echo # postdump_binlog_filename: $postdump_binlog_filename
+  --die Master state didn't change after mariadb-dump with --delete-master-logs.
+}
+
+if ($predump_binlog_pos == $postdump_binlog_pos)
+{
+  --echo # predump_binlog_pos: $predump_binlog_pos
+  --echo # postdump_binlog_pos: $postdump_binlog_pos
+  --die Master state didn't change after mariadb-dump with --delete-master-logs.
+}
+
+if ($postdump_first_binary_log_filename != $postdump_binlog_filename)
+{
+  --echo # postdump_first_binary_log_filename: $postdump_first_binary_log_filename
+  --echo # postdump_binlog_filename: $postdump_binlog_filename
+  --die Master state didn't change after mariadb-dump with --delete-master-logs.
```

Comment:
This assertion is to test that the old logs were deleted rather than the master log state changing, so this message should be changed to be clear about that.

Reply:
Sure, doing it now.